titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/util/RelocationPredicates.java

```diff
@@ -57,60 +57,60 @@
         }
 
         if (isRelocationRequired(task)) {
-            return Optional.of("Task tagged for relocation");
+            return Optional.of("Task killed because it was marked for relocation, and this was a safe time per this job's Disruption Budget");
```

Comment:
I am not sure if this message is used only when the task is actually terminated, or in the termination plan as well. It might be misleading in the latter case. Checking that task must be relocated does not mean that its disruption budget is not violated.

Reply:
We decided to stop using the word `kill` and instead use the word `terminate`.

Reply:
Oh. Yea if it is only "tagged" (not sure what that means), then that would make sense, because there are other code paths where it is *actually* terminated.

Reply:
But If I'm reading the way this works, the plan is a kind of queue of reasons (`.right()`) so I think once it is tagged, this reason is shown when it is terminated too.

Reply:
Please, have a look at the `TaskRelocationPlan` class. This is persisted in the database for tasks with self managed disruption budgets.

Reply:
Hmm. I guess I saw the `Immediate` functions above and assumed that they handled the cases where the Budget could not be respected. Are you sure this is not the part of the code that puts thing into the queue to be relocated that are *not* SelfManaged and *not* immediately relocated (and therefore, relocated due to budget?)?

> This is persisted in the database for tasks with self managed disruption budgets.

But if it is self-managed, this function returns empty right away on line 46?

Reply:
The purpose of `RelocationPredicates` is to identify tasks that require relocation. The message should reflect it. We could create a different message when evicting the task in the relocation service.

Reply:
Gotcha. I understand now that it isn't as simple as I thought it was.
I'm going to discard this as is, because I agree it will be misleading.